src/metrics.cpp

```diff
@@ -225,12 +265,24 @@ int printStats(bool mining)
 
     if (IsInitialBlockDownload(Params())) {
         int netheight = currentHeadersHeight == -1 || currentHeadersTime == 0 ? 
-            0 : EstimateNetHeight(Params().GetConsensus(), currentHeadersHeight, currentHeadersTime);
+            0 : EstimateNetHeight(params, currentHeadersHeight, currentHeadersTime);
         int downloadPercent = height * 100 / netheight;
         std::cout << "     " << _("Downloading blocks") << " | " << height << " / ~" << netheight << " (" << downloadPercent << "%)" << std::endl;
     } else {
         std::cout << "           " << _("Block height") << " | " << height << std::endl;
     }
+
+    auto nextHeight = NextActivationHeight(height, params).get_value_or(0);
+    auto nextBranch = NextEpoch(height, params).get_value_or(0);
```

Comment:
`.get_value_or(0)` is not what we should use here, because 0 is a valid return value for both functions (`NextActivationHeight` returns 0 as `ALWAYS_ACTIVE` and `NextEpoch` returns 0 for the Sprout upgrade).

After `SecondsLeftToHeight` is refactored into `SecondsLeftToNextEpoch`, move both of these lines into the else branch below, and change them to use `.get()` (because if `SecondsLeftToNextEpoch` returns some duration, both of these are guaranteed to return some value).